### PR TITLE
reset cache after heal

### PR DESCRIPTION
**Pull request by matkt**
## PR description

We found that sometimes after the heal we still have old snapshot in the cache because after the clearTrieLog some transaction pool thread are still running 
The fix is to clear the cache after the snapsync and also add a snapshot of the  new head

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java [hunk 1]
```diff
@@ -338,6 +338,9 @@ public TrieLogManager getTrieLogManager() {
   @Override
   public void setArchiveStateUnSafe(final BlockHeader blockHeader) {
```
Comment (garyschulte) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java:339]: this method name should change to reflect its use.  IMO it isn't setting the archive unsafe for use like the name implies, but rather is resetting the archive cache and adding the new pivot as the only entry.  Perhaps `resetArchiveStateTo(BlockHeader)`

Similarly, BonsaiWorldState.setArchiveUnsafe should be something more along the lines of `resetWorldStateTo(BlockHeader)`

These methods could use javadoc describing the intent

**Review by garyschulte — CHANGES_REQUESTED** [review]
LGTM, naming and documenting intent is probably wise considering the effort went into making bonsai more dev friendly

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java (hunk 1 above)
Comment (matkt) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java:339]: updated the code

**Review by garyschulte — APPROVED** [review]
🚢

**Pull request merged**